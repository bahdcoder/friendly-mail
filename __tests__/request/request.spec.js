const Request = require('../../src/Request')

describe('The Request class', () => {
  it('can be instantiated properly', () => {
    expect(new Request()).toMatchSnapshot()
  })

  it('can build a request', () => {
    const request = new Request()

    request
      .acceptJson()
      .basicAuth({ username: 'username', password: 'password' })
      .auth('token')
      .headers({
        'api-version': '2019-08-09'
      })

    expect(request._isJson).toBe(true)
    expect(request._headers).toEqual({
      'api-version': '2019-08-09'
    })
    expect(request._auth).toEqual('token')
    expect(request._basicAuth).toEqual({
      username: 'username',
      password: 'password'
    })
  })

  it('can post a request to an endpoint with all data on object', async () => {
    const request = new Request()
    request
      .acceptJson()
      .basicAuth({ username: 'username', password: 'password' })
      .auth('token')
      .headers({
        'api-version': '2019-08-09'
      })

    const response = await request.post('http://fullstackjs.online', {})

    expect(response.url).toEqual('http://fullstackjs.online')
    expect(response.data.headers).toEqual({
      Authorization: 'token',
      'api-version': '2019-08-09'
    })
  })

  it('can reject errors if post request fails', async () => {
    const request = new Request()

    expect(
      request.post('', { fail: true })
    ).rejects.toThrowErrorMatchingSnapshot()
  })
})
