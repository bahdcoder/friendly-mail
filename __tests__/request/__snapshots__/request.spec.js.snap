// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Request class can be instantiated properly 1`] = `
Request {
  "_auth": null,
  "_basicAuth": null,
  "_headers": Object {},
  "_isJson": false,
}
`;

exports[`The Request class can reject errors if post request fails 1`] = `""`;
